third_party/nvidia/backend/compiler.py

```diff
@@ -76,7 +82,12 @@ def __post_init__(self):
                "num_warps must be a power of 2"
 
     def hash(self):
-        key = '_'.join([f'{name}-{val}' for name, val in self.__dict__.items()])
+        hash_dict = dict(self.__dict__)
+        hash_dict['extern_libs'] = tuple(
+            (k, file_hash(v))
+            for k, v in hash_dict['extern_libs']
+        )
+        key = '_'.join([f'{name}-{val}' for name, val in hash_dict.items()])
```

Comment:
similarly should we sort this, so we're not relying on all the keys being added in exactly the same order?